google_shopping_functions.js

```diff
@@ -5,12 +5,38 @@ var getItemsCount = function(itemData){
 	return itemData.items.length;
 };
 
+
 //
 //DEFINE FUNCTIONS HERE
 //
+var getItems = function(objectData) {
+	return objectData.items;
+}
 
-
-
+var getItemsByBrand = function(items, brand) {
+	var objectBrands = [];
+	for (var i=0; i < items.length; i++) {
+		if(items[i].product.brand.toLowerCase() == brand.toLowerCase()) {
+			objectBrands.push(items[i]);
+		}
+	} return objectBrands;
+}
+var getItemsByAuthor = function(items, author) {
+	var objectAuthors = [];
+	for (var i=0; i < items.length; i++) {
+		if(items[i].product.author.name.toLowerCase().indexOf(author.toLowerCase()) != -1) {
+			objectAuthors.push(items[i]);
+		}
+	} return objectAuthors;
+}
+var getAvailableItems = function(items) {
+	var objectAvail = [];
+	for (var i=0; i < items.length; i++) {
+		if(items[i].product.inventories[0].availability == "inStock") {
+			objectAvail.push(items[i]);
+		}
+	} return objectAvail;
+}
 
```

Comment:
These all check out! Let's just put the return statement on a new line to keep in line with conventional formatting practices.
```
var getAvailableItems = function(items) {
	var objectAvail = [];
	for (var i = 0; i < items.length; i++) {
		if (items[i].product.inventories[0].availability == "inStock") {
			objectAvail.push(items[i]);
		}
	} 
        return objectAvail;
}
```